devito/ops/compiler.py

```diff
@@ -107,31 +102,77 @@ def _compile_cuda(self, soname):
                 sleep_delay=sleep_delay,
                 object=True
             )
-            cuda_host_compiler.link_extension(
-                '%s%s' % (self.target, cuda_host_compiler.so_ext),
-                [src_o, cuda_o],
-                debug=configuration['debug-compiler']
-            )
+            if not os.path.exists(src_o):
+                raise ValueError("Error when compiling for host\
+                                 Couldn't find file: %s" % src_o)
+
+            try:
+                cuda_host_compiler.link_extension(
+                    '%s%s' % (self.target, cuda_host_compiler.so_ext),
+                    [src_o, cuda_o],
+                    debug=configuration['debug-compiler']
+                )
```

Comment:
Same as above.

Reply:
The `_compile` function shouldn't be in CompilerOPS. Each target architecture must have its implementation.